perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathWatcherFactory.java

```diff
@@ -0,0 +1,107 @@
+package name.mitterdorfer.perlock;
+
+import name.mitterdorfer.perlock.impl.util.Preconditions;
+
+import java.nio.file.Path;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+/**
+ * Factory to create {@link PathWatcher}s that only deal with a single file and
+ * ignore change to other files within the directory
+ */
+public class SinglePathWatcherFactory {
```

Comment:
I got you now. This makes sense. And yes, `isRegularFile` will return `false` for non-existing files. However, I thought this was just intended for watching for changes in existing files but obviously this won't work when watching for file creation. So just disregard my comments related to this check. :)